backend/typescript/services/implementations/EntityService.ts

```diff
@@ -0,0 +1,40 @@
+import { userInfo } from "node:os";
+import { Entity, IEntity } from "../../models/entity.mgmodel";
+import { IEntityService, EntityRequestDTO } from "../interfaces/IEntityService";
+var mongoose = require('mongoose');
+
+
+export class EntityService implements IEntityService {
+
+    /* retrieve the Entity with the given id */
+    async getEntity(id: string): Promise<IEntity> {
+        let entity: IEntity | null;
+        entity = await Entity.findById(mongoose.Types.ObjectId(id));
+
+        if (!entity) {
+            throw new Error;
+        }
+        return entity;
+    }
+
+    /* retrieve all Entities (pagination is nice-to-have future feature) */
+    async getEntities(): Promise<IEntity[]> {
+        return await Entity.find();
```

Comment:
Could `find()` throw an error? I think a try-catch is needed

Reply:
Just realized this isn't too clear, you'll need to wait for the logger to handle errors here (in #6), let me know if it'll help if I separate out the logger into its own PR and get that merged first

Reply:
entity.find() is a mongo function that returns null/empty array if no records are found. Would a try-catch still be necessary?

Reply:
Yeah I think this might be ok for an empty query, but generally `find()` can throw (error arg in the callback) and I'd rather just be safe
https://mongoosejs.com/docs/api.html#model_Model.find